unstructured/ingest/connector/sharepoint.py

```diff
@@ -0,0 +1,190 @@
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING, List
+
+from unstructured.file_utils.filetype import EXT_TO_FILETYPE
+from unstructured.ingest.interfaces import (
+    BaseConnector,
+    BaseConnectorConfig,
+    BaseIngestDoc,
+    ConnectorCleanupMixin,
+    IngestDocCleanupMixin,
+    StandardConnectorConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+if TYPE_CHECKING:
+    from office365.sharepoint.files.file import File
+    from office365.sharepoint.publishing.pages.page import SitePage
+
+MAX_MB_SIZE = 512_000_000
+
+
+@dataclass
+class SimpleSharepointConfig(BaseConnectorConfig):
+    client_id: str
+    client_credential: str = field(repr=False)
+    site_url: str
+    folder: str
+    process_pages: bool = False
+    recursive: bool = False
+
+    def __post_init__(self):
+        if not (self.client_id and self.client_credential and self.site_url):
+            raise ValueError(
+                "Please provide one of the following mandatory values:"
+                "\n-ms-client_id\n-ms-client_cred\n-ms-sharepoint-site",
+            )
+
+
+@dataclass
+class SharepointIngestDoc(IngestDocCleanupMixin, BaseIngestDoc):
+    config: SimpleSharepointConfig
+    file: "File"
+    meta: "SitePage"
+
+    def __post_init__(self):
+        self.ext = "".join(Path(self.file.name).suffixes) if self.meta is None else ".html"
+        if not self.ext:
+            raise ValueError("Unsupported file without extension.")
+
+        if self.ext not in EXT_TO_FILETYPE.keys():
+            raise ValueError(
+                f"Extension not supported. "
+                f"Value MUST be one of {', '.join([k for k in EXT_TO_FILETYPE if k is not None])}.",
+            )
+        self._set_download_paths()
+
+    def _set_download_paths(self) -> None:
+        """Parses the folder structure from the source and creates the download and output paths"""
+        download_path = Path(f"{self.standard_config.download_dir}")
+        output_path = Path(f"{self.standard_config.output_dir}")
+        parent = (
+            Path(self.file.serverRelativeUrl[1:])
+            if self.meta is None
+            else Path(self.meta.get_property("Url", "")).with_suffix(self.ext)
+        )
+        self.download_dir = (download_path / parent.parent).resolve()
+        self.download_filepath = (download_path / parent).resolve()
+        oname = f"{str(parent)[:-len(self.ext)]}.json"
+        self.output_dir = (output_path / parent.parent).resolve()
+        self.output_filepath = (output_path / oname).resolve()
+
+    @property
+    def filename(self):
+        return Path(self.download_filepath).resolve()
+
+    @property
+    def _output_filename(self):
+        return Path(self.output_filepath).resolve()
+
+    def _get_page(self):
+        try:
+            content_labels = ["CanvasContent1"]
```

Comment:
what is the hardcoded CanvasContent1 here? is this something that will exist for every page? or is this just something you're testing as you deveop?

Reply:
(okay, I'm guessing the latter, but confirm)

Reply:
oh guessing this is something internal to define where the content of the document is. is there ever a CanvasContent2? or anything else we need to look for here?

Reply:
either way, definitely good to have a docstring and comments that explain what's going on throughout this